lib/galaxy/datatypes/microarrays.py

```diff
@@ -125,3 +126,66 @@ def set_meta(self, dataset, **kwd):
         dataset.metadata.number_of_optional_header_records = int(headers[1][0])
         dataset.metadata.number_of_data_columns = int(headers[1][1])
         dataset.metadata.file_type = headers[2][0].strip().strip('"').split("=")[1]
+
+
+class Cel(Binary):
+    """ Cel File format described at:
+            http://media.affymetrix.com/support/developer/powertools/changelog/gcos-agcc/cel.html
+    """
+
+    file_ext = "cel"
+    edam_format = "format_1638"
+    edam_data = "data_3110"
+    MetadataElement(name="version", default="3", desc="Version", readonly=True, visible=True,
+                    optional=True, no_value="3")
+
+    def sniff(self, filename):
+        """
+        Try to guess if the file is a Cel file.
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('affy_v_agcc.cel')
+        >>> Cel().sniff(fname)
+        True
+        >>> fname = get_test_fname('affy_v_3.cel')
+        >>> Cel().sniff(fname)
+        True
+        >>> fname = get_test_fname('test.cel')
+        >>> Cel().sniff(fname)
+        True
+        >>> fname = get_test_fname('test.gal')
+        >>> Cel().sniff(fname)
+        False
+        """
+        handle = open(filename, 'rb')
+        found_cel_4 = False
+        found_cel_3 = False
+        found_cel_agcc = False
+        header_bytes = handle.read(5)
+        if header_bytes.decode("utf8", errors="ignore")[0] == '@':
+            found_cel_4 = True
+        elif header_bytes.decode("utf8", errors="ignore")[0] == ';':
```

Comment:
Can you keep the previous test, which seems more specific?
```suggestion
        elif header_bytes == b';\x01\x00\x00':
```